--- index.orig	2015-06-04 14:48:11.553553374 +0200
+++ index.js	2015-06-04 14:48:29.877553514 +0200
@@ -112,14 +112,14 @@
 		search.topicSave(topicData);
 	};
 
-	search.topicDelete = function(tid, callback) {
+	search.topicDelete = function(topicData, callback) {
 		callback = callback || function() {};
 		async.parallel({
 			topic: function(next) {
-				db.searchRemove('topic', tid, next);
+				db.searchRemove('topic', topicData.tid, next);
 			},
 			posts: function(next) {
-				topics.getPids(tid, function(err, pids) {
+				topics.getPids(topicData.tid, function(err, pids) {
 					if (err) {
 						return next(err);
 					}
