--- ../../src/threadTools.orig	2015-05-29 21:04:29.113666800 +0200
+++ ../../src/threadTools.js	2015-05-29 21:04:40.965667122 +0200
@@ -38,12 +38,6 @@
 				}
 				topicData.deleted = isDelete ? 1 : 0;
 
-				if (isDelete) {
-					plugins.fireHook('action:topic.delete', tid);
-				} else {
-					plugins.fireHook('action:topic.restore', topicData);
-				}
-
 				var data = {
 					tid: tid,
 					cid: topicData.cid,
@@ -51,6 +45,13 @@
 					uid: uid
 				};
 
+				if (isDelete) {
+					plugins.fireHook('action:topic.delete', data);
+				} else {
+					plugins.fireHook('action:topic.restore', topicData);
+				}
+
+
 				callback(null, data);
 			});
 		});
